core/src/blocks/curl.rs

```diff
@@ -0,0 +1,239 @@
+        hasher.update(self.url.as_bytes());
+        hasher.update(self.headers_code.as_bytes());
+        hasher.update(self.body_code.as_bytes());
+        format!("{}", hasher.finalize().to_hex())
+    }
+
+    async fn execute(&self, _name: &str, env: &Env) -> Result<Value> {
+        let method = match self.method.as_str() {
+            "GET" => Method::GET,
+            "POST" => Method::POST,
+            "PUT" => Method::PUT,
+            "PATCH" => Method::PATCH,
+            _ => Err(anyhow!(
+                "Invalid method {}, supported methods are GET, POST, PUT.",
+                self.method
+            ))?,
+        };
+
+        let e = env.clone();
+        let headers_code = self.headers_code.clone();
+        let headers_value: Value = match tokio::task::spawn_blocking(move || {
+            let mut script = Script::from_string(headers_code.as_str())?
+                .with_timeout(std::time::Duration::from_secs(10));
+            script.call("_fun", (&e,))
+        })
+        .await?
+        {
+            Ok(v) => v,
+            Err(e) => Err(anyhow!("Error in headers code: {}", e))?,
+        };
+
+        let e = env.clone();
+        let body_code = self.body_code.clone();
+        let body_value: Value = match tokio::task::spawn_blocking(move || {
+            let mut script = Script::from_string(body_code.as_str())?
+                .with_timeout(std::time::Duration::from_secs(10));
+            script.call("_fun", (&e,))
+        })
+        .await?
+        {
+            Ok(v) => v,
+            Err(e) => Err(anyhow!("Error in body code: {}", e))?,
+        };
+
+        let url = replace_variables_in_string(&self.url, "url", env)?;
+        let parsed_url = Url::parse(url.as_str())?;
+
+        match parsed_url.scheme() {
+            "https" => (),
+            _ => Err(anyhow!("Only the `https` scheme is authorized."))?,
+        }
+
+        let _: Vec<std::net::Ipv4Addr> = match parsed_url.host() {
+            Some(h) => match h {
+                Host::Domain(d) => {
+                    let ipv4: Vec<std::net::Ipv4Addr> = lookup_host(d)?
+                        .into_iter()
+                        .filter_map(|ip| match ip {
+                            std::net::IpAddr::V4(ip) => Some(ip),
+                            _ => None,
+                        })
+                        .collect::<Vec<_>>();
+                    match ipv4.len() {
+                        0 => Err(anyhow!("Could not find an ipv4 address for host: {}", d))?,
+                        _ => ipv4,
+                    }
+                }
+                Host::Ipv4(ip) => vec![ip],
+                Host::Ipv6(_) => Err(anyhow!("Ipv6 addresses are not supported."))?,
+            },
+            None => Err(anyhow!("Provided URL has an empty host"))?,
+        }
+        .into_iter()
+        .map(|ip| {
+            lazy_static! {
+                static ref RE: Regex = Regex::new(r"^(0|127|10|172|192)\..*").unwrap();
```

Comment:
I think it's not a good idea to do string comparison here. For example, many programs interpret a leading 0 as an octal number, so 012.x.x.x is actually 10.x.x.x. If you are going to try to do the filtering this way, I would at least compare to a number, not a string.

Reply:
Can you create a example of that happening? I believe rely on the interface of the rust net package here.

Reply:
It's agree it's unlikely that it can be hit, but why not just call `.octets()[0]` and compare the numbers exactly to be sure? Regexes and string compares tend to be where security code often falters, in my (rather limited) experience.